### PR TITLE
Testing /network/clock

### PR DESCRIPTION
**Pull request by piotr-iohk**
# Issue Number

#1394

# Overview

- 82b01bc7df06ca89a26ab7c9a4c999bbd00faf86
  address that network/clock is not available on Windows at this stage
  
- 0f7c759ae15bb505b87a8c9d4039d1e25d9bdb3a
  use pendingOnWindows for network/clock integration test
  
- d49e8de165da61c1712bfe7174ee71ab05becdaf
  revise log error levels

# Comments

### CODE REVIEW

File: lib/core-integration/src/Test/Integration/Scenario/API/Network.hs [hunk 1]
```diff
@@ -199,15 +201,20 @@ spec = do
...
+        eventually "ntp status = (un)available" $ do
             r <- request @ApiNetworkClock ctx
                 Link.getNetworkClock Default Empty
             expectResponseCode @IO HTTP.status200 r
-            verify r
-                [ expectField (#ntpStatus . #status)
-                    (`shouldBe` NtpSyncingStatusAvailable)
-                ]
+            -- network/clock at this point is not supported on Windows
+            if os == "windows"
```
Comment (KtorZ) [inline lib/core-integration/src/Test/Integration/Scenario/API/Network.hs]: What? How? o.O? There's no point having this if it's no supported on Windows.

File: lib/core/src/Network/Ntp.hs [hunk 2]
```diff
@@ -118,16 +118,16 @@ instance DefineSeverity NtpTrace where
...
         NtpTraceSocketOpen _ -> Debug
         NtpTraceSocketClosed _ -> Debug
         NtpTracePacketSent _ -> Debug
-        NtpTracePacketSentError _ _ -> Alert
-        NtpTracePacketDecodeError _ _-> Alert
+        NtpTracePacketSentError _ _ -> Debug
+        NtpTracePacketDecodeError _ _-> Debug
         NtpTracePacketReceived _ -> Debug
-        NtpTraceWaitingForRepliesTimeout _ -> Alert
+        NtpTraceWaitingForRepliesTimeout _ -> Info
```
Comment (KtorZ) [inline lib/core/src/Network/Ntp.hs]: It feels a bit weird to have errors reported as Debug and Info. The timeout should at least be a `Notice` IMO, and the Error at least `Error`? What's the rationale behind this change?

File: lib/core-integration/src/Test/Integration/Scenario/API/Network.hs (hunk 1 above)
Comment (piotr-iohk) [reply lib/core-integration/src/Test/Integration/Scenario/API/Network.hs]: My understanding from discussion with @paweljakubas is that it is currently not supported on Windows in the underneath https://github.com/input-output-hk/ouroboros-network/tree/master/ntp-client, but there are plans to support it in future... So, we can either not have it, or have it and explicitly say that Windows not supported, or use different library (?)

File: lib/core/src/Network/Ntp.hs (hunk 2 above)
Comment (piotr-iohk) [reply lib/core/src/Network/Ntp.hs]: Currently there are some errors reported (from the underneath lib) on successful request -> https://github.com/input-output-hk/cardano-wallet/issues/1394#issuecomment-594496618. :man_shrugging:

File: lib/core-integration/src/Test/Integration/Scenario/API/Network.hs (hunk 1 above)
Comment (rvl) [reply lib/core-integration/src/Test/Integration/Scenario/API/Network.hs]: Maybe drop the if statement and use `Test.Utils.Windows.pendingOnWindows` here then.

Comment (piotr-iohk) [reply lib/core-integration/src/Test/Integration/Scenario/API/Network.hs]: done in 0f7c759

File: lib/core/src/Network/Ntp.hs (hunk 2 above)
Comment (KtorZ) [reply lib/core/src/Network/Ntp.hs]: These two "Alert" could / should be turned to `Notice` or `Warning` IMO. `Info` or `Debug` is a big extreme.

Comment (rvl) [reply lib/core/src/Network/Ntp.hs]: Agree.

The code tries multiple servers and fails if there are less that 3 results returned.

So log messages relating to an individual server failing to query (e.g. ipv6 misconfiguration) could be warning (or notice if you're feeling generous). It's not completely failed yet at this point.

If the whole ntpQuery fails due to too few results, it will trace NtpTraceNoLocalAddr, NtpTraceIPv4IPv6NoReplies or NtpTraceReportPolicyQueryFailed. These are proper Errors, which prevent the API endpoint from returning a result.

Comment (piotr-iohk) [reply lib/core/src/Network/Ntp.hs]: On my machine, these 2 gave me Alerts in logs on successful request:

NtpTraceRunProtocolError
NtpTracePacketSentError

```
[cardano-wallet.ntp-client:Alert:46] [2020-03-06 09:49:44.90 UTC] ntp client experienced error Network.Socket.ByteString.sendManyTo: does not exist (Network is unreachable) when sending packet using IPv6
[cardano-wallet.ntp-client:Alert:40] [2020-03-06 09:49:44.90 UTC] ntp client experienced error Network.Socket.ByteString.sendManyTo: does not exist (Network is unreachable) when using IPv6 protocol
```

Turned them into Warnings. @rvl suggested, that the errors could be genuine -> https://github.com/input-output-hk/cardano-wallet/issues/1394#issuecomment-595557011), so disabled ipv6 and, indeed, warnings disappeared.

In the end this is how it looks like:

```
		NtpTraceRunProtocolSuccess _ -> Debug
		NtpTraceSocketOpen _ -> Debug
		NtpTraceSocketClosed _ -> Debug
		NtpTracePacketSent _ -> Debug
		NtpTracePacketReceived _ -> Debug
		NtpTraceStartNtpClient -> Info
		NtpTraceTriggerUpdate -> Info
		NtpTraceRestartDelay _ -> Info
		NtpTraceRestartingClient -> Info
		NtpTraceClientSleeping -> Info
		NtpTraceClientStartQuery -> Info
		NtpTraceIPv4IPv6NoReplies -> Info
		NtpTraceQueryResult _ -> Info
		NtpTraceRunProtocolNoResult _ -> Info
		NtpTraceRunProtocolError _ _ -> Warning
		NtpTracePacketSentError _ _ -> Warning
		NtpTracePacketDecodeError _ _-> Warning
		NtpTraceWaitingForRepliesTimeout _ -> Warning
		NtpTraceIOError _ -> Alert
		NtpTraceLookupServerFailed _ -> Alert
		NtpTraceNoLocalAddr -> Alert
		NtpTraceReportPolicyQueryFailed -> Alert
```
Pls suggest if there something that should be changed.

Comment (KtorZ) [reply lib/core/src/Network/Ntp.hs]: So, on second thought looking at all these cases with more care, all the `Alert` ought to to `Warning` or `Notice`? I mean, `Alert` pretty much means: wake people up in the middle of the night to go look at what's going on with the application. 

Here, it's really just the NTP check that didn't got a response from the central NTP server. Nothing's really alarming with that, the wallet can run just fine :)

Comment (paweljakubas) [reply lib/core/src/Network/Ntp.hs]: agree that `Notice` is the best fit here as it is the closest to the definition of Notice  `needs attention; something not progressing properly`

Comment (paweljakubas) [reply lib/core/src/Network/Ntp.hs]: `Alert` is by definition much more severe : `a subsystem is no longer operating correctly, likely requires manual intervention`

Comment (piotr-iohk) [reply lib/core/src/Network/Ntp.hs]: Allrighty, turned `Alert` and `Warning` to `Notice`:

```
        NtpTraceRunProtocolSuccess _ -> Debug
        NtpTraceSocketOpen _ -> Debug
        NtpTraceSocketClosed _ -> Debug
        NtpTracePacketSent _ -> Debug
        NtpTracePacketReceived _ -> Debug
        NtpTraceStartNtpClient -> Info
        NtpTraceTriggerUpdate -> Info
        NtpTraceRestartDelay _ -> Info
        NtpTraceRestartingClient -> Info
        NtpTraceClientSleeping -> Info
        NtpTraceClientStartQuery -> Info
        NtpTraceIPv4IPv6NoReplies -> Info
        NtpTraceQueryResult _ -> Info
        NtpTraceRunProtocolNoResult _ -> Info
        NtpTraceRunProtocolError _ _ -> Notice
        NtpTracePacketSentError _ _ -> Notice
        NtpTracePacketDecodeError _ _-> Notice
        NtpTraceWaitingForRepliesTimeout _ -> Notice
        NtpTraceIOError _ -> Notice
        NtpTraceLookupServerFailed _ -> Notice
        NtpTraceNoLocalAddr -> Notice
        NtpTraceReportPolicyQueryFailed -> Notice
```

**Review by paweljakubas — APPROVED** [review]
looks fine now
